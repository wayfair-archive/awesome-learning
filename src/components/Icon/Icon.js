import React from 'react';
import './Icon.scss';

const Icon = ({ icon, cssClasses = "" }) => (
  <svg className={`Icon ${cssClasses}`} viewBox={icon.viewBox}>
    <path d={icon.path} />
  </svg>
);

export default Icon;
