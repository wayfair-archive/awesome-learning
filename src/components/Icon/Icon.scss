.Icon {
  display: block;
  width: 1em;
  height: 1em;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
  font-style: normal;
  font-weight: normal;
  speak: none;
  text-align: center;
  font-variant: normal;
  text-transform: none;
  line-height: 1em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}