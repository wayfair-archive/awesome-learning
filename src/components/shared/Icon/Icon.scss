.Icon {
  display: block;
  width: 1.5rem;
  height: 1.5rem;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
  font-style: normal;
  font-weight: normal;
  speak: none;
  text-align: center;
  font-variant: normal;
  text-transform: none;
  line-height: 1.5rem;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
