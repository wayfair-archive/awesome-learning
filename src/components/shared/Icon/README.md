Displays an Icon currently from our Icon library.
The `Icon` component takes in a prop `icon` which determines which icon to display.
See usage below

## USAGE

```
  <Icon
    icon={getIcon("logo")}
    cssClasses="Header-icon Header-icon--logo"
  />
```

## Props

The following props may be passed to configure the Block.

| name           | type                                                                                                                             | description                                  | default |
| -------------- | -------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------- | ------- |
| **icon**       | `oneOf([twitter,github, email, logo, menu, functions, array, async, data, testing, function, hooks, leftChevron, rightChevron])` | The icon to display                          |         |
| **cssClasses** | `String`                                                                                                                         | List of css classes to provide extra styling | `null`  |
