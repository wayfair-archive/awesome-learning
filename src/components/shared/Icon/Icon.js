import React from 'react';
import PropTypes from 'prop-types';
import './Icon.scss';

const Icon = ({icon, cssClasses = ''}) => (
  <svg className={`Icon ${cssClasses}`} viewBox={icon.viewBox}>
    <path d={icon.path} />
  </svg>
);

Icon.propTypes = {
  icon: PropTypes.shape({
    path: PropTypes.string.isRequired,
    viewBox: PropTypes.string,
  }).isRequired,
};

export default Icon;
