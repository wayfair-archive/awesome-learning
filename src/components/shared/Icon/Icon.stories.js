import React from 'react';
import {storiesOf} from '@storybook/react';
import {withKnobs} from '@storybook/addon-knobs';
import IconReadme from './README.md';
import {getIcon} from '../../../utils';
import Icon from './Icon';

const containerStyle = {
  display: 'flex',
  justifyContent: 'flex-start',
  flexWrap: 'wrap',
};

const blockStyle = {
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  width: '130px',
  height: '130px',
  padding: '30px',
  margin: '10px',
  border: '1px #000 solid',
};

const stories = storiesOf('Shared.Icon', module);

stories.addDecorator(withKnobs);
stories.addParameters({
  readme: {
    // Show readme at the addons panel
    sidebar: IconReadme,
  },
});

stories.add(
  'basic',
  () => (
    <div style={containerStyle}>
      <div style={blockStyle}>
        <p>Twitter</p>
        <Icon icon={getIcon('twitter')} />
      </div>
      <div style={blockStyle}>
        <p>Github</p>
        <Icon icon={getIcon('github')} />
      </div>
      <div style={blockStyle}>
        <p>Email</p>
        <Icon icon={getIcon('email')} />
      </div>
      <div style={blockStyle}>
        <p>Awesome Learning</p>
        <Icon icon={getIcon('logo')} />
      </div>
      <div style={blockStyle}>
        <p>Menu</p>
        <Icon icon={getIcon('menu')} />
      </div>
      <div style={blockStyle}>
        <p>Functions</p>
        <Icon icon={getIcon('functions')} />
      </div>
      <div style={blockStyle}>
        <p>Array</p>
        <Icon icon={getIcon('array')} />
      </div>
      <div style={blockStyle}>
        <p>Async</p>
        <Icon icon={getIcon('async')} />
      </div>
      <div style={blockStyle}>
        <p>Data</p>
        <Icon icon={getIcon('data')} />
      </div>
      <div style={blockStyle}>
        <p>Testing</p>
        <Icon icon={getIcon('testing')} />
      </div>
      <div style={blockStyle}>
        <p>Function</p>
        <Icon icon={getIcon('function')} />
      </div>
      <div style={blockStyle}>
        <p>Hook</p>
        <Icon icon={getIcon('hook')} />
      </div>
      <div style={blockStyle}>
        <p>Left Chevron</p>
        <Icon icon={getIcon('leftChevron')} />
      </div>
      <div style={blockStyle}>
        <p>Right Chevron</p>
        <Icon icon={getIcon('rightChevron')} />
      </div>
    </div>
  ),
  {info: {disable: true}}
);
