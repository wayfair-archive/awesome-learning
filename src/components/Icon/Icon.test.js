import React from 'react';
import renderer from 'react-test-renderer';
import Icon from './Icon';

describe('Icon', () => {
  const props = {
    icon: {
      viewBox: '0 0 0 0',
      path: '',
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Icon {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
