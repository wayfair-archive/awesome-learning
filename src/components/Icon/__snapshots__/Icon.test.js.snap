// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<svg
  className="Icon "
  viewBox="0 0 0 0"
>
  <path
    d=""
  />
</svg>
`;
